dev/NavigationView/NavigationView.cpp

```diff
@@ -3086,17 +3086,78 @@ void NavigationView::UpdateTopNavListViewItemSource(const winrt::IInspectable& i
 
 void NavigationView::UpdateListViewItemSource()
 {
-    if (!m_appliedTemplate)
-    {
-        return;
-    }
-
     auto dataSource = MenuItemsSource();
     if (!dataSource)
     {
         dataSource = MenuItems();
     }
 
+    auto iterable = dataSource.try_as<winrt::IIterable<winrt::IInspectable>>();
```

Comment:
Can you make your assert as a function? also I expect something like #if BUILD_WINDOWS, just skip the check.

Reply:
There's no assert here. Did you mean the entire validation logic? I didn't put it into a function because it's only called in this one place.

It probably should be under !BUILD_WINDOWS though. No need to run this logic at all in that situation.